rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala

```diff
@@ -84,8 +84,13 @@ object RholangCLI {
 
   private def normalizeTerm(term: Proc, inputs: ProcVisitInputs) = {
     val normalizedTerm = ProcNormalizeMatcher.normalizeMatch(term, inputs)
-    if (normalizedTerm.par.freeCount > 0)
-      throw new Error("Top-level free variables are not allowed.")
+    if (normalizedTerm.par.freeCount > 0) {
```

Comment:
Note that this interacts with your wildcard PR.
You can still check the knownFree map for the same thing, and that might be easier.
We should probably record the locations where wildcards occurred.

Reply:
I'll take care of this in the other PR and I'll open a third for the wildcards.